### ISSUE TITLE
Unusable in Angular since typescript update to v5.1

### ISSUE THREAD

**Issue by AlxBrnk** [baumerdev/ibantools-germany#15]
## Expected Behavior
iban tools validation functions should be usable with Angular v16.0 and lower.

## Actual Behavior
`main.js:1 ERROR TypeError: Cannot read properties of undefined (reading 'VALID')
    at a.isValidAccountNumberBLZ (main.js:1:252620)`

## Steps to Reproduce the Problem
1. create new angular app of version v15 or lower
2. import and call any of the provided iban validation methods
3. start app with production build `ng serve --configuration production`

## Specifications
- Version: 1.2301.3+
- Platform: Angular < 16.1.5

Since the typescript update to v5.1 with version 1.2301.3 any Angular application that is older than [16.1.5](https://github.com/angular/angular-cli/releases/tag/16.1.5) will have this issue, as the Angular compiler will break any ts enums if they are typescript 5.1 based and compiled to a commonjs module.
https://github.com/angular/angular-cli/issues/25524

It seems to be an issue if the only available build target is commonjs.
Would it be possible to add additional module targets (es6/amd) to the dist or downgrade to typescript v5 again?

**Comment by baumerdev** [comment]
Since this seems to only be a problem with the Angular compiler's buildOptimizer, so I'm hesitating to downgrade the dependency and hold it back for x days/months/years until a t.b.d./sufficient adoption rate for Angular 16.1.5 is reached.

Another option that comes to my mind is replacing the enum in question
```ts
export enum Result {
  INVALID = "INVALID",
  //...
}
```
with a const like
```ts
export const Result = {
  INVALID: "INVALID",
  //...
} as const;
```
which would still allow for syntax like Result.INVALID ... but on the other hand needs a type change in params and returns from `Result` to `keyof typeof Result`. Not sure if this would be fully downward compatible to put this in a minor/fix update.

So I'm currently thinking about your proposal of adding the ne build with 
```json
"compilerOptions": {
  "target": "ES6",
  "module": "AMD"
}
```
but I'm not sure how I can integrate this in the package.json's `exports` next to the cjs and esm ?!

**Comment by baumerdev** [comment]
@AlxBrnk Can you check out branch **poc_remove_ts_enum** of this repo, build it and tell me if it works in your Angular environment? As mentioned in #16 this'd be a version without enums in TypeScript 1393e5904be6548a59a4220c1bf22af7e63c39a9.

**Comment by AlxBrnk** [comment]
@baumerdev  It works 👍🏻

**Comment by baumerdev** [comment]
I just published v1.2301.5 to npmjs, so this issue should be be considered closed.

… and hopefully no-one will open a follow-up because these changes broke something different 🤞
